datumaro/plugins/ndr.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+import cv2
+import numpy as np
+from scipy.linalg import orth
+
+from datumaro.components.extractor import Transform, DEFAULT_SUBSET_NAME
+
+class NDR(Transform):
+    """
+    Near-duplicated image removal |n
+    Removes near-duplicated images in subset |n
+    """
+    def __init__(self, extractor, working_subset=None, duplicated_subset="duplicated", algorithm='gradient',
+            num_cut=None, over_sample='random', under_sample='uniform', seed=None, **kwargs):
+        """
+        Near-duplicated image removal
+
+        Arguments
+        ---------------
+        working_subset: str
+            name of the subset to operate
+            if None, use DEFAULT_SUBSET_NAME
+        duplicated_subset: str
+            name of the subset for the removed data after NDR runs
+        algorithm: str
+            name of the algorithm to use
+            "gradient" only for now.
+        num_cut: int
+            number of outputs you want.
+            the algorithm will cut whole dataset to this amount
+            if None, return result without any modification
+        over_sample: "random" or "similarity"
+            specify the strategy when num_cut > length of the result after removal
+            if random, sample from removed data randomly
+            if similarity, select from removed data with ascending order of similarity
+        under_sample: "uniform" or "inverse"
+            specify the strategy when num_cut < length of the result after removal
+            if uniform, sample data with uniform distribution
+            if inverse, sample data with reciprocal of the number of data which have same hash key
+        Algorithm Specific for gradient
+            block_shape: tuple, (h, w)
+                for the robustness, this function will operate on blocks
+                mean and variance will be calculated on this block
+            hash_dim: int
+                dimension(or bit) of the hash function
+            sim_threshold: float
+                the threshold value for saving hash-collided samples.
+                larger value means more generous, i.e., saving more samples
+        Return
+        ---------------
+        None, other subsets combined with the result
+        """
+        super().__init__(extractor)
+        if seed:
+            np.random.seed(seed)
+        if len(extractor) < 1:
```

Comment:
I don't think this situation should be considered an error.

Reply:
Okay, I will update